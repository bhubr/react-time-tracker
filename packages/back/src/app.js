// src/app.js
import './env';
import express from 'express';
import bodyParser from 'body-parser';
import path from 'path';
import basicAuth from 'express-basic-auth';

import tasksRouter from './routes/tasks';
import timeSlicesRouter from './routes/timeSlices';
import lockRouter from './routes/lock';

const app = express();

app.use(bodyParser.json());
app.use('/api/tasks', tasksRouter);
app.use('/api/time-slices', timeSlicesRouter);
app.use('/api/lock', lockRouter);

const { NODE_ENV, AUTH_USER, AUTH_PASS } = process.env;

// Enforce basic auth if AUTH_* vars are set
if (AUTH_USER && AUTH_PASS) {
  app.use(basicAuth({
    challenge: true,
    users: { [AUTH_USER]: AUTH_PASS }
  }));
}

// Check whether we are in production env
const isProd = NODE_ENV === 'production';

if (isProd) {
  // Compute the build path and index.html path
  const buildPath = path.resolve(__dirname, '../../front/build');
  const indexHtml = path.join(buildPath, 'index.html');

  // Setup build path as a static assets path
  app.use(express.static(buildPath));
  // Serve index.html on unmatched routes
  app.get('*', (req, res) => res.sendFile(indexHtml));
}

export default app;
