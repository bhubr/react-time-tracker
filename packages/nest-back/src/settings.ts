import { ConnectionOptions } from 'typeorm';

const {
  NODE_ENV,
  DATABASE_URL,
  DB_HOST,
  DB_NAME,
  DB_USER,
  DB_PASS,
  DB_SYNCHRONIZE
} = process.env;
const isProd = NODE_ENV === 'production';
const isTest = NODE_ENV === 'test';

let connSettings: ConnectionOptions;

// A bug in TypeORM prevents getting the db name from the url
// https://github.com/typeorm/typeorm/issues/2096
// Waiting for it to be fixed
// if (isProd) {
//   connSettings = {
//     type: 'mysql',
//     url: DATABASE_URL,
//     synchronize: DB_SYNCHRONIZE === 'true',
//   };
// } else if (isTest) {
if (isTest) {
  connSettings = {
    type: 'mysql',
    port: 3306,
    host: 'localhost',
    username: 'trakttest',
    password: 'trakttest',
    database: 'trakt_test',
    synchronize: true,
  };
} else {
  connSettings = {
    type: 'mysql',
    port: 3306,
    host: DB_HOST,
    username: DB_USER,
    password: DB_PASS,
    database: DB_NAME,
    synchronize: !isProd || DB_SYNCHRONIZE === 'true',
  };
}

export default {
  database: connSettings
};
