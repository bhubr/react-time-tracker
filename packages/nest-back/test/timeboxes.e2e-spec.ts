import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication } from '@nestjs/common';
import { createConnection, Connection } from 'typeorm';
import * as request from 'supertest';
import { AppModule } from '../src/app.module';
import settings from '../src/settings';

describe('TimeboxController (e2e)', () => {
  let app: INestApplication;
  let connection: Connection;

  beforeAll(async () => {
    connection = await createConnection({ ...settings.database, name: 'testconn' });
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();
  });

  beforeEach(async () => {
    await connection.query('SET FOREIGN_KEY_CHECKS=0;');
    await connection.query('TRUNCATE TABLE timebox');
    await connection.query('TRUNCATE TABLE task');
    await connection.query('SET FOREIGN_KEY_CHECKS=1;');
  });

  it('POST /api/timeboxes SUCCESS', async () => {
    const resTask = await request(app.getHttpServer())
      .post('/api/tasks')
      .send({ title: 'Parent Task #1' })
      .expect(201);
    const task = resTask.body;

    const resTimebox = await request(app.getHttpServer())
      .post('/api/timeboxes')
      .send({ taskId: task.id, comment: 'Timebox #1.1', start: '2020-01-07 08:00', type: 'POMODORO' })
      .expect(201);
    expect(resTimebox.body).toMatchSnapshot();
  });

});