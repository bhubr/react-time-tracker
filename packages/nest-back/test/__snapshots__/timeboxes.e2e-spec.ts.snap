// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeboxController (e2e) POST /api/timeboxes SUCCESS 1`] = `
Object {
  "comment": "Timebox #1.1",
  "end": null,
  "id": 1,
  "start": "2020-01-07 08:00",
  "taskId": 1,
  "type": "POMODORO",
}
`;
